packages/wallet/src/redux/initialized/reducer.ts

```diff
@@ -28,6 +32,39 @@ const combinedReducer = combineReducersWithSideEffects({
   fundingState: fundingStateReducer,
 });
 
-export function coordinator(state: InitializedState, action: WalletAction): InitializedState {
-  return state;
+export function coordinator(
+  state: InitializedState,
+  action: actions.WalletAction,
+): InitializedState {
+  switch (action.type) {
+    case actions.funding.FUNDING_RECEIVED_EVENT:
+      return fundingReceivedEventCoordinator(state, action);
+    default:
+      return state;
+  }
+}
+
+function fundingReceivedEventCoordinator(
+  state: InitializedState,
+  action: actions.funding.FundingReceivedEvent,
+): InitializedState {
+  const { channelId } = action;
+  const channelStatus = getChannelStatus(state, channelId);
+  if (channelStatus.type !== WAIT_FOR_FUNDING_AND_POST_FUND_SETUP) {
+    return state;
+  }
+  const fundingStatus = getDirectFundingStatus(state, channelId);
+
+  const newState = { ...state };
+  if (stateIsChannelFunded(fundingStatus)) {
+    const { state: newChannelStatus, sideEffects } = fundingReducer(
```

Comment:
This might be an explicit transition, in which case we can completely get rid of "internal actions"